### PR TITLE
[Gutenberg] Upgrade React Native `0.71.11`

### PR DESCRIPTION
**Pull request by fluiddot**
**Related to https://github.com/wordpress-mobile/gutenberg-mobile/pull/5874**

**To test:**
Follow testing instructions from https://github.com/WordPress/gutenberg/pull/51303.

## Regression Notes
1. Potential unintended areas of impact
None apart from the editor.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and CI tests.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [ ] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -1,6 +1,6 @@
 pluginManagement {
     gradle.ext.kotlinVersion = '1.8.21'
-    gradle.ext.agpVersion = '7.2.1'
+    gradle.ext.agpVersion = '7.3.1'
```
Comment (fluiddot) [inline settings.gradle]: ⚠️ As discussed internally (p1686309448014639-slack-CC7L49W13), for now, we won't upgrade AGP. Hence, we should revert this change.

**Comment by fluiddot** [comment]
Some of the CI jobs are failing due to [a recent commit](https://github.com/wordpress-mobile/WordPress-Android/commit/8c60d496f40686b9943cd311d9981f96d5c68722) merged into `trunk`, not the changes incorporated in this PR. It will be solved via https://github.com/wordpress-mobile/WordPress-Android/pull/18733.

**UPDATE:** This issue has been fixed 🎊 .

File: libs/editor/build.gradle [hunk 2]
```diff
@@ -68,25 +68,18 @@ dependencies {
...
+    implementation "com.facebook.react:hermes-android:$facebookReactVersion"
+    implementation "com.facebook.react:react-android:$facebookReactVersion"
 
     // This dependency will be substituted if the `local-builds.gradle` file contains
     // `localGutenbergMobilePath`. Details for this can be found in the `settings.gradle` file.
     implementation ("$rootProject.gradle.ext.gutenbergMobileBinaryPath:$rootProject.ext.gutenbergMobileVersion") {
+        // We exclude React Native modules from older and newer versions.
+        // This is needed to avoid Duplicate class errors when building using local Gutenberg Mobile path.
         exclude module: 'hermes-engine'
         exclude module: 'react-native'
```
Comment (fluiddot) [inline libs/editor/build.gradle:80]: In theory, we don't need these modules from the previous RN version (i.e. `0.69.4`). However, I'm experiencing Duplicate class errors when building if I remove them. Seems that the fact that some of the React Native third-party libraries ([especially the forked repositories](https://github.com/wordpress-mobile/react-native-video/blob/197c644d6d1b86d0ad86831310992e71b776c9b3/android/build.gradle#L34)) are still pointing to RN 0.69.4, has to do with this issue.

**Review by SiobhyB — APPROVED** [review]
Tested and confirmed all looks good 🙌

**Review by dcalhoun — APPROVED** [review]
